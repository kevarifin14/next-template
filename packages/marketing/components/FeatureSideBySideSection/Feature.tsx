import { IconType } from 'react-icons';

export type IFeature = {
  name: string,
  description: string,
  icon: IconType,
};

type FeatureProps = {
  feature: IFeature,
};

export default function Feature({ feature }: FeatureProps) {
  return (
    <div key={feature.name} className="relative">

      <dt>
        <div className="absolute flex items-center justify-center h-12 w-12 rounded-md bg-primary text-white">
          <feature.icon className="h-6 w-6" />
        </div>
        <h3 className="ml-16 m-0">{feature.name}</h3>
      </dt>

      <p className="mt-2 ml-16">{feature.description}</p>
    </div>

  );
}
