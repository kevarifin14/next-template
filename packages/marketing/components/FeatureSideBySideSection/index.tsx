import Section from 'components/Section';

import Feature, { IFeature } from './Feature';

type FeatureSideBySideSectionProps = {
  title: string,
  src: string,
  description?: string,
  features: IFeature[],
  reverse?: boolean,
};

export default function FeatureSideBySideSection({
  title, description, features, src, reverse,
}: FeatureSideBySideSectionProps) {
  return (
    <Section size="7xl" className="lg:grid lg:grid-cols-2 lg:gap-8 lg:items-center">
      <div className={reverse ? 'col-start-2' : ''}>

        <h1 className="font-extrabold tracking-tight m-0 mb-2">
          {title}
        </h1>

        <p className="m-0">
          {description}
        </p>

        <dl className="mt-10 space-y-10">
          {features.map((feature) => <Feature feature={feature} />)}
        </dl>

      </div>

      <div className={reverse ? 'col-start-1 row-start-1' : ''}>
        <img className="relative mx-auto mt-16 lg:mt-0" width={490} src={src} alt="" />
      </div>
    </Section>
  );
}
