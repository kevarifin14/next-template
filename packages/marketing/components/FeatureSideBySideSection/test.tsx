import { render } from '@testing-library/react';

import {
  mockDescription, mockFeatures, mockSrc, mockTitle,
} from '__mocks__/utils';

import FeatureSideBySideSection from '.';

describe('FeatureSideBySideSection', () => {
  const renderFeatureSideBySideSection = (props = {}) => (
    render(
      <FeatureSideBySideSection
        title={mockTitle}
        description={mockDescription}
        features={mockFeatures}
        src={mockSrc}
        {...props}
      />,
    )
  );

  it('matches snapshot', () => {
    const { baseElement } = renderFeatureSideBySideSection();
    expect(baseElement).toMatchSnapshot();
  });

  it('reverses', () => {
    const { container } = renderFeatureSideBySideSection({ reverse: true });

    expect(container.firstChild.firstChild).toHaveClass('col-start-2');
    expect(container.firstChild.lastChild).toHaveClass('col-start-1 row-start-1');
  });
});
