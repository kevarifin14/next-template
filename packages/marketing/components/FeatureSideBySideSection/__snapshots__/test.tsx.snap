// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeatureSideBySideSection matches snapshot 1`] = `
<body>
  <div>
    <div
      class="py-8 px-4 lg:py-12 lg:px-6 w-full prose dark:prose-light mx-auto max-w-7xl lg:grid lg:grid-cols-2 lg:gap-8 lg:items-center"
    >
      <div
        class=""
      >
        <h1
          class="font-extrabold tracking-tight m-0 mb-2"
        >
          mockTitle
        </h1>
        <p
          class="m-0"
        >
          mockDescription
        </p>
        <dl
          class="mt-10 space-y-10"
        >
          <div
            class="relative"
          >
            <dt>
              <div
                class="absolute flex items-center justify-center h-12 w-12 rounded-md bg-primary text-white"
              >
                <svg
                  class="h-6 w-6"
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3.055 11H5a2 2 0 012 2v1a2 2 0 002 2 2 2 0 012 2v2.945M8 3.935V5.5A2.5 2.5 0 0010.5 8h.5a2 2 0 012 2 2 2 0 104 0 2 2 0 012-2h1.064M15 20.488V18a2 2 0 012-2h3.064M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </div>
              <h3
                class="ml-16 m-0"
              >
                Marketing
              </h3>
            </dt>
            <p
              class="mt-2 ml-16"
            >
              Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores impedit perferendis suscipit eaque, iste dolor cupiditate blanditiis ratione.
            </p>
          </div>
          <div
            class="relative"
          >
            <dt>
              <div
                class="absolute flex items-center justify-center h-12 w-12 rounded-md bg-primary text-white"
              >
                <svg
                  class="h-6 w-6"
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M9.75 17L9 20l-1 1h8l-1-1-.75-3M3 13h18M5 17h14a2 2 0 002-2V5a2 2 0 00-2-2H5a2 2 0 00-2 2v10a2 2 0 002 2z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </div>
              <h3
                class="ml-16 m-0"
              >
                Application
              </h3>
            </dt>
            <p
              class="mt-2 ml-16"
            >
              Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores impedit perferendis suscipit eaque, iste dolor cupiditate blanditiis ratione.
            </p>
          </div>
          <div
            class="relative"
          >
            <dt>
              <div
                class="absolute flex items-center justify-center h-12 w-12 rounded-md bg-primary text-white"
              >
                <svg
                  class="h-6 w-6"
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M13 10V3L4 14h7v7l9-11h-7z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </div>
              <h3
                class="ml-16 m-0"
              >
                Design System
              </h3>
            </dt>
            <p
              class="mt-2 ml-16"
            >
              Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores impedit perferendis suscipit eaque, iste dolor cupiditate blanditiis ratione.
            </p>
          </div>
        </dl>
      </div>
      <div
        class=""
      >
        <img
          alt=""
          class="relative mx-auto mt-16 lg:mt-0"
          src="mockSrc"
          width="490"
        />
      </div>
    </div>
  </div>
</body>
`;
